.github/workflows/sanitize.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow builds & runs test cases in iceoryx
+
+name: Sanitize
+
+# Triggers the workflow on push or pull request events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:  
+  clang-sanitize:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-18.04
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name: Install iceoryx dependencies
+        # Softwares installed in ubuntu-18.04 instance
+        # https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1804-README.md
+        run: sudo apt-get update && sudo apt-get install -y libacl1-dev libncurses5-dev
+
+      - name : Checkout
+        uses: actions/checkout@v2
+
+      - name : Clang Sanitize
+        run: |
+          export CC=/usr/bin/clang
+          export CXX=/usr/bin/clang++
+          export NUM_CORES=`nproc`
+          # build tests , dont run them
+          $GITHUB_WORKSPACE/tools/iceoryx_build_test.sh build-test sanitize clean -j $NUM_CORES
```

Comment:
Any reasons the dds gateway is not included? With the removal of the clang build, the dds gateway wouldn't be build with clang anymore

Reply:
> nice, you will make @elfenpiff quite happy with this

I hope I wont make anybody unhappy with this ;-)

Reply:
I skipped it because there were no tests. You are right , it would be miss clang build.
Its added

Reply:
yes, it's more like ensuring it's building with clang and there are also no warnings. But you are right, we should probably also execute the tests with the clang build